### DISCUSSION TITLE
Using `@apply` in Tailwind plugins

### DISCUSSION THREAD

**Discussion by babakfp** [tailwindlabs/tailwindcss#9336] [Help] [answered]
Hi
Can we have the ability to use the `@apply` keyword in our custom Tailwind plugins? It will make the whole thing easy. If I want to use the class `transform` (in my custom plugin), I need to copy paste all of its styles, instead of just using adding that class! We are used to using Tailwind classes, so let us use it in our custom plugins too. 

Something like this:

```js
const plugin = require('tailwindcss/plugin')

module.exports = plugin(({ addComponents }) => {
  addComponents({
    '.my-class': {
      '@apply': 'p-4 bg-gray-100 rounded',
    },
  })
})
```

That's all. Thanks.

**Comment by wongjn** [comment, marked as answer]
You can already do this, we just need to adjust your code a little bit:

```diff
   addComponents({
     '.my-class': {
-      '@apply': 'p-4 bg-gray-100 rounded',
+      '@apply p-4 bg-gray-100 rounded': {},
     },
   })
```

See [this Tailwind Play for a working example](https://play.tailwindcss.com/rWbRD9GPaq?file=config)

**Reply by babakfp** [reply]
👍Thanks
I couldn't find it on the docs🤷‍♂️
